jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -580,7 +582,7 @@ public MethodHandle findSpecial(Class<?> clazz, String methodName, MethodType ty
 					/*[MSG "K0586", "Lookup class ({0}) must be the same as or subclass of the current class ({1})"]*/
 					throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K0586", accessClass, handleDefc)); //$NON-NLS-1$
 				}
-				checkAccess(handle);
+				checkAccess(handle, false);
```

Comment:
don't MethodHandle lookups also get the implied read access?

Reply:
`j.l.MethodHandles.findSpecial()` performs `checkSpecialAccess(specialToken)` which ensures `PRIVATE access` or `accessClass equals callerClass` hence `checkClassModuleVisibility() Objects.equals(accessModule, targetModule)` is true regardless of `reflectiveAccess`. In this case specifying `true` or `false` in this `checkAccess(handle, reflectiveAccess)` won't make a difference.

On the other hand, not all `MethodHandle lookups` have the implied read access, for example, `j.l.MethodHandles.findVirtual()` in following code snippet doesn't imply read access to `unnamed` modules:
```
Class<?> lookupTarget = Class.forName("class.to.be.loaded.from.classpath");
MethodHandles.lookup().findVirtual(lookupTarget, "testFindSpecial", MethodType.methodType(void.class));
``` 
This should throw `java.lang.IllegalAccessException: 'test.pkg1' no access to: 'test.pkg2'`.

Reply:
@tajila does comment above address your concern?